src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -1124,15 +1124,33 @@ int Simulator::publish_distance_topic(const mavlink_distance_sensor_t *dist_mavl
 	dist.orientation = dist_mavlink->orientation;
 	dist.variance = dist_mavlink->covariance * 1e-4f; // cm^2 to m^2
 	dist.signal_quality = -1;
-
 	dist.h_fov = dist_mavlink->horizontal_fov;
 	dist.v_fov = dist_mavlink->vertical_fov;
 	dist.q[0] = dist_mavlink->quaternion[0];
 	dist.q[1] = dist_mavlink->quaternion[1];
 	dist.q[2] = dist_mavlink->quaternion[2];
 	dist.q[3] = dist_mavlink->quaternion[3];
 
-	_dist_pub.publish(dist);
+	// Create 1st publisher instance if it doesn't exist
+	if (_dist_pubs[0] == nullptr) {
```

Comment:
This is not needed.

Reply:
@bkueng I tried to remove this but it seems that when I don't force an init on the first index, it doesn't init the publisher at all (I get a  `never published`).